src/metpy/calc/thermo.py

```diff
@@ -647,7 +647,7 @@ def _parcel_profile_helper(pressure, temperature, dewpt):
 
 def _insert_lcl_level(pressure, temperature, lcl_pressure):
     """Insert the LCL pressure into the profile."""
-    interp_temp = interpolate_1d(lcl_pressure, pressure, temperature)
+    interp_temp = interpolate_1d(np.round(lcl_pressure, 4), pressure, temperature)
```

Comment:
We have some helpers like `_greater_or_close`, etc. Is there a fix for this problem that involves using those helpers in `interpolate_1d`? Or is that outside our control (e.g. because of `np.sort`). If we need to `round()`, should we be doing it inside `interpolate_1d`?

Reply:
I can look and see if any of these options would take care of it